clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -568,19 +572,30 @@ public static ByteBuffer ensureCapacity(ByteBuffer existingBuffer, int newLength
      *
      * @param file The root file at which to begin deleting
      */
-    public static void delete(File file) {
-        if (file == null) {
+    public static void delete(final File file) throws IOException {
```

Comment:
Maybe we should rename this to `deleteIfExists` since we don't throw an error if `file` doesn't exist. Probably in a subsequent PR if that makes sense to you.